### PR TITLE
S3 Object Storage - SSE-C / SSE-S3 encryption - Update

### PR DESCRIPTION
**Pull request by OVH-Team-Guides**
Replacing Pull Request #5950 from @KebirHakim 

This pull request updates the French documentation located at pages/storage_and_backup/object_storage/s3_encrypt_your_objects_with_sse_c_and_sse_s3/guide.fr-fr.md. The draft s3_encrypt_your_objects_with_sse_c has been finalized and is now available. 

The updated content provides detailed guidance on implementing server-side encryption with SSE-C (Server-Side Encryption with Customer Keys) and SSE-S3 (Server-Side Encryption with S3-Managed Keys) for object storage on OVHcloud S3.

This enhancement aims to improve security measures and user understanding of encryption options available on OVHcloud S3.

### CODE REVIEW

File: pages/storage_and_backup/object_storage/s3_encrypt_your_objects_with_sse_c/guide.fr-fr.md [hunk 1]
```diff
@@ -148,7 +165,251 @@ Les URL pré-signées, qui peuvent être utilisées pour des opérations telles
...
+
+#### Avantages
+
+- **Gestion des Clés Simplifiée**
+
+Avec OVHcloud prenant en charge la gestion sécurisée des clés de chiffrement, les utilisateurs bénéficient d'une gestion des clés simplifiée. Cette approche élimine les complexités liées à la rotation des clés tout en maintenant un niveau élevé de sécurité pour les données. Elle permet de concilier sécurité et efficacité opérationnelle, en éliminant la charge administrative de la gestion manuelle des clés de chiffrement.
+
+- **Sécurité Renforcée**
+
+Nous employons une stratégie de chiffrement avancée pour offrir une protection maximale de vos données. Chaque bucket bénéficie d'une clé unique, et pour chaque objet stocké, une clé de chiffrement distincte est générée. Cette clé est obtenue en combinant la clé unique du bucket avec un sel aléatoire, ce qui assure que chaque objet est chiffré avec sa propre clé. Cette méthode de dérivation de clés limite le risque associé à l'exposition d'une clé unique et garantit une sécurité renforcée pour vos données.
```
Comment (OVH-Team-Guides) [inline pages/storage_and_backup/object_storage/s3_encrypt_your_objects_with_sse_c/guide.fr-fr.md:181]: `avec un sel aléatoire`
wrong translation?

Comment (KebirHakim) [reply pages/storage_and_backup/object_storage/s3_encrypt_your_objects_with_sse_c/guide.fr-fr.md:181]: The term "sel aléatoire" in the context of encryption refers to the practice of adding random data when generating an encryption key, to ensure that each key is unique, even if the main key (or the unique key of the bucket) is reused.

File: pages/storage_and_backup/object_storage/s3_encrypt_your_objects_with_sse_c/guide.fr-fr.md [hunk 2]
```diff
@@ -148,7 +165,251 @@ Les URL pré-signées, qui peuvent être utilisées pour des opérations telles
...
+- `<region>` : remplacez par la région de votre service OVHcloud S3, qui héberge votre bucket.
+
+L'option `--server-side-encryption AES256` dans la commande indique que vous souhaitez appliquer le chiffrement SSE-S3. Cela garantit que l'objet envoyé est chiffré de manière sécurisée directement sur le serveur OVHcloud, offrant une couche supplémentaire de protection pour vos données.
+
+##### Téléchargement d'un Objet avec SSE-S3 sur OVHcloud S3
+
+Pour télécharger un objet qui a été chiffré avec SSE-S3 depuis OVHcloud S3, il n'est pas nécessaire de spécifier des headers du chiffrement dans la commande. En effet, l'objet peut être téléchargé directement sans manipulation supplémentaire liée au chiffrement, car le déchiffrement est géré automatiquement côté serveur. Voici un exemple de commande de téléchargement :
+
+```bash
+aws s3api get-object --bucket votre-bucket --key votre-objet chemin/vers/destination/fichier --endpoint-url https://s3.<region>.ovhcloud.com
```
Comment (lason-ovh) [inline pages/storage_and_backup/object_storage/s3_encrypt_your_objects_with_sse_c/guide.fr-fr.md]: wrong endpoint: use https://s3.<region>.io.cloud.ovh.net

File: pages/storage_and_backup/object_storage/s3_encrypt_your_objects_with_sse_c/guide.fr-fr.md [hunk 3]
```diff
@@ -148,7 +165,251 @@ Les URL pré-signées, qui peuvent être utilisées pour des opérations telles
...
+- Le paramètre `--endpoint-url https://s3.<region>.ovhcloud.com` doit être ajusté à la région de votre service OVHcloud S3.
+
+Attention de ne pas inclure de headers de chiffrement spécifiques lors du téléchargement d'un objet chiffré avec SSE-S3 pour éviter des erreurs, telles qu'une erreur 400 Bad Request. 
+
+#### Ajout du chiffrement à un bucket existant sur OVHcloud S3
+
+Pour ajouter le chiffrement SSE-S3 à un bucket S3 existant sur OVHcloud, vous devez utiliser la commande `put-bucket-encryption` de l'AWS CLI. Cette commande configure le chiffrement du bucket pour que tous les nouveaux objets ajoutés soient automatiquement chiffrés avec SSE-S3. Voici la commande spécifique que vous utiliseriez :
+
+```bash
+aws s3api put-bucket-encryption --bucket votre-bucket --server-side-encryption-configuration '{"Rules":[{"ApplyServerSideEncryptionByDefault":{"SSEAlgorithm":"AES256"}}]}' --endpoint-url https://s3.<region>.ovhcloud.com
```
Comment (lason-ovh) [inline pages/storage_and_backup/object_storage/s3_encrypt_your_objects_with_sse_c/guide.fr-fr.md]: wrong endpoint: use https://s3.<region>.io.cloud.ovh.net instead

File: pages/storage_and_backup/object_storage/s3_encrypt_your_objects_with_sse_c/guide.fr-fr.md [hunk 4]
```diff
@@ -148,7 +165,251 @@ Les URL pré-signées, qui peuvent être utilisées pour des opérations telles
...
+- Remplacez `<region>` par la région de votre service OVHcloud S3 où votre bucket est situé.
+
+Cela va configurer le bucket pour utiliser le chiffrement SSE-S3 avec les clés gérées par S3 (AES256) pour tous les nouveaux objets. Les objets existants ne seront pas affectés. Si vous souhaitez également les chiffrer, vous devrez les copier ou les uploader à nouveau après avoir changé cette configuration.
+
+##### Affichage de la configuration du chiffrement du Bucket
+
+Après avoir configuré le chiffrement de votre bucket via `PutBucketEncryption` pour utiliser SSE-S3, assurez-vous que tout est correctement mis en place en utilisant la commande suivante avec l'AWS CLI :
+
+```bash
+aws s3api get-bucket-encryption --bucket votre-bucket --endpoint-url https://s3.<region>.ovhcloud.com
```
Comment (lason-ovh) [inline pages/storage_and_backup/object_storage/s3_encrypt_your_objects_with_sse_c/guide.fr-fr.md]: wrong endpoint: use https://s3.<region>.io.cloud.ovh.net instead

File: pages/storage_and_backup/object_storage/s3_encrypt_your_objects_with_sse_c/guide.fr-fr.md [hunk 5]
```diff
@@ -148,7 +165,251 @@ Les URL pré-signées, qui peuvent être utilisées pour des opérations telles
...
+L'implémentation du chiffrement SSE-S3 sur OVHcloud S3 est conçue pour offrir une gestion des clés de chiffrement à la fois sécurisée et transparente pour l'utilisateur. Chaque bucket bénéficie d'une clé distincte, ce qui assure que les données y sont sécurisées de façon individuelle et isolée. Cette méthode de chiffrement, intégrée et gérée par la plateforme, élimine les complexités associées à la gestion manuelle des clés par les utilisateurs. Tout en rendant le processus utilisateur aussi fluide et intuitif que possible, elle maintient une sécurité robuste et conforme aux standards les plus stricts en matière de protection des données.
+
+#### Envoi d'un objet avec SSE-S3 sur OVHcloud S3
+
+##### Upload d'un objet sur OVHcloud S3 avec chiffrement SSE-S3
+
+Pour envoyer un objet dans votre bucket S3 sur OVHcloud avec chiffrement SSE-S3, utilisez la commande Bash suivante via l'AWS CLI. Cette commande intègre l'option de chiffrement côté serveur pour renforcer la sécurité de vos données stockées.
+
+```bash
+aws s3api put-object --bucket votre-bucket --key votre-objet --body chemin/vers/votre/fichier --server-side-encryption AES256 --endpoint-url https://s3.<region>.ovhcloud.com
```
Comment (lason-ovh) [inline pages/storage_and_backup/object_storage/s3_encrypt_your_objects_with_sse_c/guide.fr-fr.md]: wrong endpoint: use https://s3.<region>.io.cloud.ovh.net instead

File: pages/storage_and_backup/object_storage/s3_encrypt_your_objects_with_sse_c/guide.fr-fr.md [hunk 6]
```diff
@@ -148,7 +165,251 @@ Les URL pré-signées, qui peuvent être utilisées pour des opérations telles
...
+Pour télécharger un objet qui a été chiffré avec SSE-S3 depuis OVHcloud S3, il n'est pas nécessaire de spécifier des headers du chiffrement dans la commande. En effet, l'objet peut être téléchargé directement sans manipulation supplémentaire liée au chiffrement, car le déchiffrement est géré automatiquement côté serveur. Voici un exemple de commande de téléchargement :
+
+```bash
+aws s3api get-object --bucket votre-bucket --key votre-objet chemin/vers/destination/fichier --endpoint-url https://s3.<region>.ovhcloud.com 
+```
+
+- Remplacez `votre-bucket` par le nom de votre bucket.
+- Remplacez `votre-objet` par la clé de l'objet que vous souhaitez télécharger.
+- Remplacez `chemin/vers/destination/fichier` par le chemin où vous souhaitez sauvegarder le fichier téléchargé.
+- Le paramètre `--endpoint-url https://s3.<region>.ovhcloud.com` doit être ajusté à la région de votre service OVHcloud S3.
```
Comment (lason-ovh) [inline pages/storage_and_backup/object_storage/s3_encrypt_your_objects_with_sse_c/guide.fr-fr.md]: wrong endpoint: use https://s3.<region>.io.cloud.ovh.net instead

**Review by lason-ovh — CHANGES_REQUESTED** [review]
wrong endpoint url (see comments)

**Comment by OVH-Team-Guides** [comment]
@lason-ovh , @KebirHakim , please review FR changes and the fully updated EN version.

**Comment by OVH-Team-Guides** [comment]
Validated. Not to merge before feature prod (ETA is 2024-04-08)

**Comment by OVH-Team-Guides** [comment]
Thank you for your contribution @KebirHakim

**Pull request merged**